packages/data/src/registry.js

```diff
@@ -171,5 +180,9 @@ export function createRegistry( storeConfigs = {} ) {
 		...storeConfigs,
```

Comment:
The line prior: We register `core/data` by default. The store tracks resolutions, which feels like something we should want to be considered globally, maybe represented by only registering it by default for the root registry. What do you think?

Reply:
yes, `core/data` is proving to be more and more challenging :). to be honest I feel we have two options:

 - Have a single global `core/data` but the question of how to differentiate between stores with similar keys and different registries will be asked.
 - Instead of having `core/data` being a store, having more like a `reducer` embedded in all stores. There's some backward compatibility concerns with this approach but what I like about it is that it makes the stores even more independent from the registries (related to this comment https://github.com/WordPress/gutenberg/pull/14367#discussion_r266977475)

Reply:
> Instead of having core/data being a store, having more like a reducer embedded in all stores. There's some backward compatibility concerns with this approach but what I like about it is that it makes the stores even more independent from the registries (related to this comment #14367 (comment))

This approach sounds interesting to me.  How would the actions on core/data be exposed (or would there be a special api for the resolution actions/selectors?)

Reply:
> How would the actions on core/data be exposed 

We could keep them expose as is (like a virtual store) or have separate actions/selectors for each store (we could even do this today as a shortcut).

If this is something you're interested in exploring, feel free to do so :)

Reply:
> If this is something you're interested in exploring, feel free to do so :)

I'd love to.  Unfortunately my contributions right now are pretty limited to grunt work/feedback/reviews as my client workload has increased.  Still I might manage to carve out some personal time to experiment.

Reply:
> The line prior: We register `core/data` by default. The store tracks resolutions, which feels like something we should want to be considered globally, maybe represented by only registering it by default for the root registry. What do you think?

Noting that in a variant of #14367 with Reusable Block implemented with an embedded `<EditorProvider />`, this proved to be quite problematic, where saving would not complete because [a `resolveSelect`](https://github.com/WordPress/gutenberg/blob/475087cff5358886183521d1382fcd57a35c4457/packages/editor/src/store/actions.js#L282-L286) would never resolve due to the fact that [resolution conditions](https://github.com/WordPress/gutenberg/blob/475087cff5358886183521d1382fcd57a35c4457/packages/editor/src/store/controls.js#L93-L96) were considered on its own copy of `core/data`, despite the fact that resolution had occurred on the inherited `core` store (from the parent, with its own different `core/data`).

I guess at the very least, `hasFinishedResolution` would need to be aware of whether the `storeKey` it receives is one defined in the current registry, or which would be inherited from its parent.

> * Have a single global `core/data` but the question of how to differentiate between stores with similar keys and different registries will be asked.

Depending on whether it's possible and without needing to impact the external API, I wonder if it's a matter of creating some unique identifier per resolver and tracking that as part of resolution status in the "global" `core/data`. I guess it'd still be challenging in that if a registry was using an inherited store, considering resolution status would have to check its parent registry instead.

Reply:
Just throwing out, I wonder if resolution state should be enhanced on registered stores?  That way its specific to the stores?

Reply:
> Just throwing out, I wonder if resolution state should be enhanced on registered store state as opposed to global state? That way its specific to the stores?

Yeah, this sounds similar to what @youknowriad suggested with the embedded reducer. On the note of backwards-compatibility, I don't think it's an issue **_if_** the idea is that we still offered selectors from `core/data` to check resolved status. The problem there is how we'd reimplement those selectors to check the states of the individual stores.

I could see it being a compatibility issue if we're saying that we'd extend the stores with additional selectors like `select( 'core/editor' ).hasFinishedResolution`, but it wasn't clear to me that's what was proposed, or what's desired.

Reply:
> > The line prior: We register `core/data` by default. The store tracks resolutions, which feels like something we should want to be considered globally, maybe represented by only registering it by default for the root registry. What do you think?
> 
> Noting that in a variant of #14367 with Reusable Block implemented with an embedded `<EditorProvider />`, this proved to be quite problematic, where saving would not complete because [a `resolveSelect`](https://github.com/WordPress/gutenberg/blob/475087cff5358886183521d1382fcd57a35c4457/packages/editor/src/store/actions.js#L282-L286) would never resolve due to the fact that [resolution conditions](https://github.com/WordPress/gutenberg/blob/475087cff5358886183521d1382fcd57a35c4457/packages/editor/src/store/controls.js#L93-L96) were considered on its own copy of `core/data`, despite the fact that resolution had occurred on the inherited `core` store (from the parent, with its own different `core/data`).

Trying to rephrase this, because in re-reading it, I don't think it's very clear:

In the embedded editor, when a save occurs, the implementation of the action will cause a selector call to `select( 'core' ).getPostType`. Since the `core` store is not explicitly registered in the child registry, it will be called against that of its parent. The resolver still takes effect, but it checks resolution status against the _parent_'s `core/data` store. Since `getPostType` would have been resolved much earlier in the parent store, the resolver never triggers its fetch. From the perspective of the child store, which has its own instance of `core/data`, the resolution still has not occurred, so it will forever become stuck.

Locally, I resolved this with the following:

```diff
diff --git a/packages/data/src/registry.js b/packages/data/src/registry.js
index 9f8268360..2f4bb41bd 100644
--- a/packages/data/src/registry.js
+++ b/packages/data/src/registry.js
@@ -175,10 +175,16 @@ export function createRegistry( storeConfigs = {}, parent = null ) {
 		return registry;
 	}
 
-	Object.entries( {
-		'core/data': dataStore,
-		...storeConfigs,
-	} ).map( ( [ name, config ] ) => registry.registerStore( name, config ) );
+	if ( parent === null ) {
+		storeConfigs = {
+			'core/data': dataStore,
+			...storeConfigs,
+		};
+	}
+
+	Object.entries( storeConfigs ).forEach( ( [ name, config ] ) => {
+		registry.registerStore( name, config );
+	} );
 
 	if ( parent ) {
 		parent.subscribe( globalListener );
```

But to @youknowriad's point above, the problem becomes that, from `core/data`, we distinguish on store name and selector, selector args, but notably *not* registry, so there's a high potential for conflict if two stores in separate registries each have their own need to resolve.

Reply:
Now that #14634 is merged, it seems that we can move forward here